core-models/src/main/scala/com/mesosphere/usi/core/models/PodSpec.scala

```diff
@@ -12,4 +16,62 @@ package com.mesosphere.usi.core.models
   * @param goal target goal of this pod. See [[Goal]] for more details
   * @param runSpec WIP the thing to run, and resource requirements, etc.
   */
-case class PodSpec(id: PodId, goal: Goal, runSpec: RunSpec)
+trait PodSpec {
```

Comment:
Independent of the rest of this PR - could we please change it back to `case class` and give a private constructor?

Reply:
I think this is how you do private constructor for case classes in scala. Stackoverflow agrees :) https://stackoverflow.com/questions/20030826/scala-case-class-private-constructor-but-public-apply-method

Reply:
Really?

```scala
case class Foo private (b: String)
Foo("bar") // passes
new Foo("bar") // fails
```
WTF. I sometimes hate Scala. Anyhow, Java won't profit from any so I propose to use the builder pattern.

Reply:
Ah, I just talked with @zen-dog. It works like this

```scala
case class Foo private (b: String)
object Foo {
  def apply(b: String) = {
    throw new NotImplementedException("I overrode the default factory method")
  }
}
Foo("bar") // throws exception
new Foo("bar") // fails
```

Reply:
My vote is that we don't make the constructor private, and we don't force validation. Let's talk about this more tomorrow in a meeting.